src/bindings/rust/elektra/src/examples.rs

```diff
@@ -0,0 +1,47 @@
+#[cfg(test)]
+mod elektra_examples {
+    use crate::{KeyBuilder, ReadableKey, StringKey, WriteableKey};
+
+    #[test]
+    fn key_example() -> Result<(), Box<dyn std::error::Error>> {
+        // This test should stay in sync with the example in the Readme
+        // To create a simple key with a name and value
+        let mut key = StringKey::new("user/test/language")?;
+        key.set_value("rust");
+        assert_eq!(key.name(), "user/test/language");
+        assert_eq!(key.value(), "rust");
+
+        // To iterate over the name
+        for name in key.name_iter() {
+            println!("Name: {}", name);
+        }
+
+        // Duplicate a key
+        let key_duplicate = key.duplicate();
+
+        // And compare them
+        assert_eq!(key, key_duplicate);
+
+        // To create a key with multiple meta values, use the KeyBuilder
+        let mut key: StringKey = KeyBuilder::new("user/test/fruits")?
+            .meta("banana", "🍌")?
+            .meta("pineapple", "🍍")?
+            .meta("strawberry", "🍓")?
+            .build();
+        assert_eq!(key.meta("pineapple")?.value(), "🍍");
+
+        // We can iterate over the metakeys
+        key.rewind_meta();
```

Comment:
It would be nice if this is not needed (copy meta-data to other key) OR make the other iterators (like the name iterator) also like this.

Reply:
Yes that would be really nice! But I'm not sure what you mean with copying metadata? Basically copy the key, `rewind_meta()` on that key and then iterate over that? 
That would work, but for the KeySet later rewinding will still be necessary.

Reply:
Yes, duplicating the whole key (set) for the iterator also works.

> That would work, but for the KeySet later rewinding will still be necessary.

Why? If the iterator duplicates the KeySet it should be fine.

Reply:
Right, because duplicating a KeySet is only a shallow copy. But with KeySets I could also store the cursor, iterate over it and then restore it in the destructor.